src/main/java/org/folio/entlinks/service/dataloader/SourceFileDeserializer.java

```diff
@@ -44,14 +45,22 @@ public AuthoritySourceFile deserialize(JsonParser parser, DeserializationContext
     var type = Optional.ofNullable(node.get("type")).map(JsonNode::asText).orElse(null);
     var name = Optional.ofNullable(node.get("name")).map(JsonNode::asText).orElse(null);
     var url = Optional.ofNullable(node.get("baseUrl")).map(JsonNode::asText).orElse(null);
-    var source = Optional.ofNullable(node.get("source")).map(JsonNode::asText).orElse(null);
+    final var source = Optional.ofNullable(node.get("source")).map(JsonNode::asText).orElse(null);
 
     sourceFile.setId(id);
     sourceFile.setName(name);
     sourceFile.setType(type);
     sourceFile.setBaseUrl(url);
-    sourceFile.setSource(source);
-
+    setSourceType(sourceFile, source);
     return sourceFile;
   }
+
+  private static void setSourceType(AuthoritySourceFile sourceFile, String source) {
+    if (source != null) {
+      switch (source) {
```

Comment:
This could be just a simple if statement as we do not have more than one case to check.